src/components/flowchart-wrapper/flowchart-wrapper.js

```diff
@@ -78,6 +83,51 @@ export const FlowChartWrapper = ({
     matchedFocusedNode,
   } = findMatchedPath(pathname, search);
 
+  /**
+   * Sets the query params from local storage based on NodeType, tag, expandAllPipelines and active pipeline.
+   * @param {string} activePipeline - The active pipeline.
+   */
+  const setParamsFromLocalStorage = (activePipeline) => {
+    const localStorageParams = loadLocalStorage(localStorageName);
+    if (localStorageParams) {
+      const paramActions = {
+        pipeline: (value) => {
+          if (!searchParams.has(params.pipeline) && activePipeline) {
+            toSetQueryParam(params.pipeline, value.active || activePipeline);
+          }
+        },
+        tag: (value) => {
+          if (!searchParams.has(params.tags)) {
+            const enabledKeys = getKeysByValue(value.enabled, true);
+            enabledKeys && toSetQueryParam(params.tags, enabledKeys);
+          }
+        },
+        nodeType: (value) => {
+          if (!searchParams.has(params.types)) {
+            const disabledKeys = getKeysByValue(value.disabled, false);
+            disabledKeys && toSetQueryParam(params.types, disabledKeys);
+          }
+        },
+        flags: (value) => {
```

Comment:
is there any particular reason this is called flags ? other keys like pipeline, tag, nodeType mimick what the query is. Can we rename this to expandAllPipeline ?

Reply:
@ravi-kumar-pilla The `flags` key in the state object(loadLocalStorage) is used to store flags(expandAllPipelines, sizewarning).
If I use expandAllPipeline instead of flags I need to make extra checks and mapping here

```
      for (const [key, value] of Object.entries(localStorageParams)) {
        if (paramActions[key]) {
          paramActions[key](value);
        }
      }
```

<img width="628" alt="Screenshot 2024-04-02 at 11 20 56 a m" src="https://github.com/kedro-org/kedro-viz/assets/38945204/472b32e5-6895-408f-88ae-bfadb517f93b">